app/src/main/java/com/android/unio/model/user/Auth.kt

```diff
@@ -98,3 +103,61 @@ fun isValidEmail(text: String): Boolean {
 fun isValidPassword(text: String): Boolean {
   return text.length in 6..4096 && text.contains(Regex("[0-9]"))
 }
+
+/**
+ * Deletes the user with the given userId from firebase auth, storage (for the profile picture) and
+ * firestore
+ *
+ * @param userId The Id of the corresponding user we want to delete
+ * @param authViewModel The instance of the authViewModel to delete the user in firebase auth
+ * @param userViewModel The instance of the userViewModel to delete the user in firestore
+ * @param imageRepository The instance of the image repository to delete the user's profile picture
+ * @return true if all three method were successful and false otherwise
+ */
+suspend fun deleteUser(
```

Comment:
Thanks, you are absolutely right, i don't know what I was thinking! I think i started to write it here because of all the different viewModels you needed, but it's pretty easy to just import the necessary repos thanks to @Hilt ;)